editor/editor_node.cpp

```diff
@@ -999,13 +999,23 @@ void EditorNode::save_resource_as(const Ref<Resource> &p_resource, const String
 	file->clear_filters();
 
 	List<String> preferred;
-	for (int i = 0; i < extensions.size(); i++) {
-		if (p_resource->is_class("Script") && (extensions[i] == "tres" || extensions[i] == "res" || extensions[i] == "xml")) {
+	for (List<String>::Element *E = extensions.front(); E; E = E->next()) {
+		if (p_resource->is_class("Script") && (E->get() == "tres" || E->get() == "res" || E->get() == "xml")) {
```

Comment:
XML resources are no longer supported since Godot 3.0:

```suggestion
		if (p_resource->is_class("Script") && (E->get() == "tres" || E->get() == "res")) {
```